### DISCUSSION TITLE
use azure OpenAI？

### DISCUSSION THREAD

**Discussion by jasonz1360** [assafelovic/gpt-researcher#165] [Ideas]
I only have azure OpenAI, so how do I integrate azure OpenAI quickly?

**Comment by LazaUK** [comment]
What exactly you want to integrate? You may quickly test Azure OpenAI model's functionality in a Studio UI, generate sample code there or re-use some samples from this page: https://learn.microsoft.com/en-us/azure/ai-services/openai/quickstart?tabs=command-line&pivots=programming-language-python

**Comment by VCSinc** [comment]
I was wondering if restarting this discussion was preferred over starting a new one. 

I have a similar query. I want to change the default LLM API call from OpenAI to use my Endpoints with Azure OpenAI Models I have. This is primarily due to my token rate limit and RPM/RPD with Azure being much higher than with OpenAI.

I was looking through the code but could not find the exact file where the Agent ChatCompletion objects were created.

**Comment by danieldekay** [comment]
This is implemented.

In your .env file, you need to add the keys as described in https://github.com/assafelovic/gpt-researcher/blob/master/docs/docs/gpt-researcher/llms.md

Or for more detailed understanding go to the source: https://github.com/assafelovic/gpt-researcher/blob/master/gpt_researcher/llm_provider/azureopenai/azureopenai.py
It's all described in the beginning of the file.

**Reply by JoeRoddy** [reply]
@danieldekay both of these resources /404 now btw.  currently this discussion is the top result on google for this topic.

**Comment by JoeRoddy** [comment]
https://docs.gptr.dev/docs/gpt-researcher/llms/running-with-azure
